playbooks/roles/ocp-install/tasks/checkcert.yaml

```diff
@@ -0,0 +1,44 @@
+---
+# tasks file for ocp4 install cert
+
+- name: check if oc is installed
+  command: type -P oc
+  register: oc_installed
+
+# check certificate validity
+# cert_valid is 1 if current time is in certificate validity time period, 0 otherwise
+- name: check the certificate validity
+  shell: |
+    dates=$(jq .ignition.security.tls.certificateAuthorities[].source openstack-upi/worker.ign |
+            sed -n -e  's/^\"data:text\/plain;charset=utf-8;base64,\(.*\)\"/\1/p' |
+            base64 --decode |
+            openssl x509 -noout -in - -dates)
+
+    notBefore=$(echo "${dates}" | sed -n -e 's/^notBefore=//p' | date -d - +%s)
+    notAfter=$(echo "${dates}" | sed -n -e 's/^notAfter=//p' | date -d - +%s)
+    present=`date +%s`
+
+    if (( $present > ${notBefore} )) && (( $present < ${notAfter} )); then
+       echo 1
+    else
+       echo 0
+    fi
+  register: cert_valid
+  when: oc_installed.rc == 0
+
+# update worker.ign file with new cert on HTTP Server if the certifiacte is invalid.
+# update to worker.ign is based on redhat solution  https://access.redhat.com/solutions/4799921
+- name: Update a certificate if required
+  shell: |
+    MCS=api-int.{{ install_config.cluster_id }}.{{ install_config.cluster_domain }}:22623
+
+    # update the tls certificate in the worker ignition file
+    echo "q"                                                        | \
+    openssl s_client -connect $MCS  -showcerts                      | \
+    awk '/-----BEGIN CERTIFICATE-----/,/-----END CERTIFICATE-----/' | \
+    base64 --wrap=0                                                 | \
+    tee api-int.base64 && sed --regexp-extended --in-place=.backup "s%base64,[^,]+%base64,$(cat api-int.base64)\"}]}}%" openstack-upi/worker.ign
```

Comment:
Can we copy the original ign file into worker.ign.orig? asking because multiple runs to add workers will keep on creating backup file over-writing the original contents.

Reply:
fair enough. I will make an original copy for safe keeping. and each run will also create a backup file for the last run.